### PR TITLE
Internal implementation of PCoA

### PR DESCRIPTION
**Pull request by yxtay**
There was no additional comments on #65 after I added the example notebook with the internal implementation of the `pcoa()` function. I take it that it is ok and have completed the rest of the necessary cleanup.

Below are the main changes:
- reimplemented `pcoa()` function to removed dependencies on `scikit-bio`.
- removed `scikit-bio` and added `scikit-learn` in the `requirements.txt` file.
- implemented another MDS method based on Metric Multidimensional Scaling from `sklearn`.
- added a notebook comparing the different MDS methods available.

Hope all is fine and can be merged soon.

### CODE REVIEW

File: pyLDAvis/_prepare.py [hunk 1]
```diff
@@ -78,8 +66,42 @@ def _jensen_shannon(_P, _Q):
...
+        eigvals[ix_neg] = np.zeros(eigvals[ix_neg].shape)
+        eigvecs[:, ix_neg] = np.zeros(eigvecs[:, ix_neg].shape)
+
+    return np.sqrt(eigvals) * eigvecs
+
+
 def js_PCoA(distributions):
-   """Dimension reduction via Jensen-Shannon Divergence & Principal Components
+    """Dimension reduction via Jensen-Shannon Divergence & Principal Coordinate Analysis
+    (oka Classical Multidimensional Scaling)
```
Comment (bmabey) [inline pyLDAvis/_prepare.py]: typo... s/oka/aka/

File: pyLDAvis/_prepare.py [hunk 2]
```diff
@@ -89,13 +111,26 @@ def js_PCoA(distributions):
...
     pcoa : array, shape (`n_dists`, 2)
-   """
-   dist_matrix = DistanceMatrix(dist.squareform(dist.pdist(distributions.values, _jensen_shannon)))
-   if skbio_old:
-       data = PCoA(dist_matrix).scores()
-       return data.site[:,0:2]
-   else:
-       return pcoa(dist_matrix).samples.values[:, 0:2]
+    """
+    dist_matrix = pairwise_distances(distributions, metric=_jensen_shannon)
```
Comment (bmabey) [inline pyLDAvis/_prepare.py]: Is there anything that sklearn's `pairwise_distances` gets you that `scipy.spatial.distance.pdist` does not?

I ask because this one function is the one function being used from `sklearn` for the default `PCoA` option. If we move back to `pdist` here then we can make `sklearn` an optional dep just like we had it before.

**Comment by bmabey** [comment]
Thanks @yxtay. Overall this looks great. I pulled it down and the tests pass so this version is verified to be similar enough to what we had.

Per my inline comments I think we should still make `skleran` and optional dep, will you make those changes please? 

WRT the notebook you added. Instead of adding a notebook just for MMDS lets just update the existing ones and delete the one you've added. I went ahead and updated the overview notebook that you can merge into your branch here: https://github.com/bmabey/pyLDAvis/commit/9361348060e1d7e0730e750f5f09f95fa6883722
I think we should also update the sklearn notebook to show the `MMDS`. When you update that notebook please use the print function so it works in Python 3.

Thanks for this!

**Comment by yxtay** [comment]
@bmabey, thanks for going through my PR and commenting on my codes. I have made the necessary changes.

Just curious, any particular reason to keep `sklearn` optional? I initially added it for the `MDS()` function, but it didn't work out as I hoped. But I think it may still be useful in the future, and will mean not having to workaround, in the codes, the possibility of users not having it installed....

**Comment by bmabey** [comment]
Thanks again @yxtay! I've merged in the pull request and have released all these changes as `v2.0.0.` Please try out latest release and let me know if you run into any issues with it.
